src/tools/pkcs11-tool.c

```diff
@@ -8185,9 +8353,9 @@ static int test_decrypt(CK_SESSION_HANDLE sess)
 #else
 		for (n = 0; n < num_mechs; n++) {
 			switch (mechs[n]) {
-			case CKM_RSA_PKCS_OAEP:
+//			case CKM_RSA_PKCS_OAEP:
 				/* one more OAEP test with param .. */
-				errors += encrypt_decrypt(sess, mechs[n], privKeyObject, "ABC", 3);
+// 				errors += encrypt_decrypt(sess, mechs[n], privKeyObject, "ABC", 3);
```

Comment:
> Why is this test commented out? Either add a comment/description why it is not used or remove it.

See:   f861829b6a07084f31e9314090f05990f18e24fa  "pkcs11-tool.c - softhsm  problem with CKM_RSA_PKCS_OAEP skip for now"

Because the test fails when using softmsm2 in the "tests" which appears to not accept the "label" of "ABC".
But Softhsm2 appears to handle a NULL label in line 8368: `errors += encrypt_decrypt(sess, mechs[n], privKeyObject, NULL, 0)`

Possible solutions:
- Do we have the test to always fail until it is fixed in softhsm2?
- Remove the test in pkcs11-tool.c?
- Change line 8364 to not increment errors, but print an warning  message? 
- Add code to `pkcs11-tool.c` when compiled for the CI to skip this line?

What would you like to see?

Reply:
I am ok with keeping it commented out, but with a comment describing why it is and ideally with a link to softhsm issue, if there is one.

Reply:
I have not submitted an issue for this problem yet. From https://github.com/opendnssec/SoftHSMv2/issues/712 it sounds like there is no development going on for the last 3 years, with 32 outstanding PRs and all original developers doing other things. Does RedHat want to get involved with maintaining SoftHsm2?

(The "label" "ABC" is actually the `CK_RSA_PKCS_OAEP_PARAMS pSourceData`.)

I would suggests changing: `errors += encrypt_decrypt(sess, mechs[n], privKeyObject, "ABC", 3);` to print a warning, and not increment `errors`. And also test if `CK_TOKEN_INFO  manufacturerID`  identifies it as softhsm.
I will look to see what it returns. I have been using 2.6.1, but need to check what version the CI is using.

Reply:
More testing and looking at the SoftHSM2 code only works with `--hash-algorithm SHA-1 and --mgf MGF1-SHA1` and only works when `pSourceData = NULL`, which is accepted  but does not used in Decrypt operation so decryption fails.  

pkcs1-tool will print: 
printf(" -- CKM_RSA_PKCS_OAEP with pSourceData != NULL, skipping\n");

See: a8126fc73a0d74c4820eb02ea2cf18d74456b6ba  in pkcs11-tool.c 

See: a8126fc73a0d74c4820eb02ea2cf18d74456b6ba in tests/test-pkcs11-tool-test.sh

Reply:
The softhsm is limited. There are some attempts to restart the development of sofhtsm2 as it is still the most widely available softhware pkcs11 module, but we will see. With recent changes to the NSS, we might even try to change the testsuite to use NSS softoken, but it is clearly outside of the scope of this PR.

I would like to avoid tying up the pkcs11-tool tests only to softhsm. It should be generic to support any pkcs11 module. 

If we will change it to match only the softhsm needs, some other pkcs11 modules might have issues with it (such as the ones not allowing SHA1 anymore).

I think the best would be to skip/comment out this test as the oaep is not that important, but we need at least the comment as outlined in the original comment.

Reply:
Can we reach some middle ground? 

`pkcs11-tool --test` with SoftHSM will always fail because only SHA-1 and MGF-1 is required and even with these it will fail with any pSourceData.  The failure in one case is by `util_fatal` so no other tests are run. 
`tests/test-pkcs11-tool-test.sh` will then fail and not do the URI tests.  One of the  changes was to `Makefile.am` to move from the "XFAIL_TESTS". 

When `pkcs11-tool --test` is run by a user, the `util_fatal` will also cause no following tests to be run. But user may  get a clue  as to how to get it to pass. 

My whole goal in changing `pkcs11-tool` for the  `CKM_RSA_PKCS_OAEP` was to get this PR to pass all the tests when used with SoftHSM. 

So what do you want:
- remove running `pkcs11-tool --test` when using SoftHSM2 
- modify `pkcs11-tool` when running `--test` to treat as error, not failure. This could be messy.
- add `test-pkcs11-tool-test.sh` back into  "XFAIL_TESTS". 
- add `|| true` to ignore the failure so URI tests are also run.
- remove the `printf(" -- CKM_RSA_PKCS_OAEP with pSourceData != NULL, skipping\n");` and let `pkcs11-tool` try the original tests. 
- Add code in pkcs11-tool to test if SoftHSM2 is the module, or if being run under CI (maybe `if (getenv("something"))`

Reply:
I am ok with either of these, but prefer skipping this only for the softhsm rather than every time. Doing

> remove running pkcs11-tool --test when using SoftHSM2

would be simplest and should not make test coverage much lower as we run p11test too with `make check`.